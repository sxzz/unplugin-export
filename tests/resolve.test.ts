import path from 'node:path'
import { build } from 'esbuild'
import glob from 'fast-glob'
import { rollup } from 'rollup'
import Esbuild from 'rollup-plugin-esbuild'
import { describe, expect, test } from 'vitest'
import EsBuildPlugin from '../src/esbuild'
import RollupPlugin from '../src/rollup'

describe('resolve', async () => {
  const fixtures = await glob(['./fixtures/*.{js,ts}', '!**/*.d.ts'], {
    absolute: true,
    cwd: __dirname,
  })
  for (const fixture of fixtures) {
    const ext = path.extname(fixture)
    const filename = path.basename(fixture, ext)

    test(`Rollup: ${filename}`, async () => {
      const bundle = await rollup({
        input: fixture,
        treeshake: false,
        plugins: [
          RollupPlugin({
            dts: path.resolve(path.dirname(fixture), `${filename}-glob`),
          }),
          Esbuild(),
        ],
      })
      const { output } = await bundle.generate({})
      expect(output[0].code).toMatchSnapshot()
    })

    test(`esbuild: ${filename}`, async () => {
      const bundle = await build({
        entryPoints: [fixture],
        treeShaking: false,
        bundle: true,
        write: false,
        plugins: [
          EsBuildPlugin({
            dts: path.resolve(path.dirname(fixture), `${filename}-glob`),
          }),
        ],
      })
      const { outputFiles } = bundle
      expect(outputFiles[0].text).toMatchSnapshot()
    })
  }
})
