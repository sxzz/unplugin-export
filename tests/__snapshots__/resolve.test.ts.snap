// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`resolve > Rollup: basic 1`] = `
"const a = "a";

const foo = "bar";

"use strict";

export { a, foo };
"
`;

exports[`resolve > Rollup: exclude-self 1`] = `
"const a = "a";

const foo = "bar";

export { a, foo };
"
`;

exports[`resolve > Rollup: import-all 1`] = `
"const a = "a";

const foo = "bar";

console.log(a);
console.log(foo);

export { a, foo };
"
`;

exports[`resolve > Rollup: ts 1`] = `
"const a = "a";

const foo = "bar";

export { a, foo };
"
`;

exports[`resolve > esbuild: basic 1`] = `
"(() => {
  // tests/fixtures/mod/a.ts
  var a = "a";

  // tests/fixtures/mod/b.ts
  var foo = "bar";
})();
"
`;

exports[`resolve > esbuild: exclude-self 1`] = `
"(() => {
  // tests/fixtures/mod/a.ts
  var a = "a";

  // tests/fixtures/mod/b.ts
  var foo = "bar";
})();
"
`;

exports[`resolve > esbuild: import-all 1`] = `
"(() => {
  // tests/fixtures/mod/a.ts
  var a = "a";

  // tests/fixtures/mod/b.ts
  var foo = "bar";

  // tests/fixtures/import-all.ts
  console.log(a);
  console.log(foo);
})();
"
`;

exports[`resolve > esbuild: ts 1`] = `
"(() => {
  // tests/fixtures/mod/a.ts
  var a = "a";

  // tests/fixtures/mod/b.ts
  var foo = "bar";
})();
"
`;
